nyansat/host/shell/command_invoker.py

```diff
@@ -0,0 +1,222 @@
+            self.exec_("config.set(\"%s\", %s)" % (key, val))
+
+    def config_get_default(self, key):
+        """Get the default value of a config parameter.
+
+        Arguments:
+        key -- name of config parameter.
+        """
+        return self.eval_string_expr("config.get_default(\"{}\")".format(key))
+
+    def config_new(self, name):
+        """Create a new config file on the ESP32.
+
+        Arguments:
+        name -- name of new config file.
+        """
+        self.exec_("config.new(\"{}\")".format(name))
+
+    def config_switch(self, name):
+        """Switch to using a different config file.
+
+        Arguments:
+        name -- name of config file.
+        """
+        self.exec_("config.switch(\"{}\")".format(name))
+
+    def i2c_scan(self, sda, scl):
+        """
+        Create and scan an i2c bus for addresses; helpful for debugging
+        :param sda: Pin number for sda
+        :param scl: Pin number for scl
+        :return: Addresses found on i2c bus
+        """
+        self.exec_("import machine")
+        self.exec_("from machine import Pin")
+        self.exec_(
+            "i2c = machine.I2C(-1, sda=Pin({}, Pin.OUT, Pin.PULL_DOWN), scl=Pin({}, Pin.OUT, Pin.PULL_DOWN))".format(
+                    sda,
+                    scl
+                )
+            )
+        return self.eval_string_expr("i2c.scan()")
+
+    def imu_calibration_status(self):
+        """Get IMU calibration status."""
+        return json.loads(self.eval_string_expr("api.imu.get_calibration_status()"))
+
+    def imu_save_calibration_profile(self):
+        """Save the current IMU calibration as 'calibration.json'."""
+        return self.eval_string_expr("api.imu.save_calibration_profile('calibration.json')")
+
+    def imu_upload_calibration_profile(self):
+        """Upload 'calibration.json' to the IMU."""
+        return self.eval_string_expr("api.imu.upload_calibration_profile('calibration.json')")
+
+    def motor_test(self, index, pos):
+        """Run a motor accuracy test, for testing the disparity between the motor and IMU.
+
+        Arguments:
+        index -- index of motor on the PWM driver (defaults: 0 == elevation, 1 == azimuth).
+        pos -- desired angle.
+        """
+        return ast.literal_eval(self.eval_string_expr("api.motor_test({}, {})".format(index, pos)))
+
+    def set_elevation_degree(self, el_angle):
+        """Set the elevation angle.
+
+        Arguments:
+        el_angle -- desired elevation angle.
+        """
+        self.eval_string_expr("api.antenna.set_elevation({})".format(el_angle))
+
+    def set_azimuth_degree(self, az_angle):
+        """Set the azimuth angle.
+
+        Arguments:
+        az_angle -- desired azimuth angle.
+        """
+        self.eval_string_expr("api.antenna.set_azimuth({})".format(az_angle))
+
+    def create_antkontrol(self):
+        """Create an antkontrol object on the ESP32."""
+        try:
+            ret = self.exec_("import antenny")
```

Comment:
I see a potential issue cropping up here; the `config` object is tied to the initialization of `antkontrol`; if `antkontrol` is started up, then `config` should be `api.config`. While most "valid" program control flow paths will result in the intended behavior, I can see race conditions make this work not as intended. We can discuss this later at length